rclpy/rclpy/executors.py

```diff
@@ -57,10 +47,13 @@ def wait(self, timeout_sec=None):
         """
         Wait until all work completes.
 
-        :param timeout_sec: Seconds to wait. Block forever if None. Don't wait if <= 0
+        :param timeout_sec: Seconds to wait. Block forever if None or negative. Don't wait if 0
         :type timeout_sec: float or None
-        :rtype: bool True if all work completed
+        :rtype: bool
+        :returns: True if all work completed
         """
+        if timeout_sec is not None and timeout_sec < 0:
+            timeout_sec = None
         # Wait for all work to complete
         if timeout_sec is None or timeout_sec >= 0:
```

Comment:
isn't this always evaluating to True ? (given the previous 2 lines)

Reply:
Good catch 6c77285b326bd4235d236d616801982f623d588f